### PR TITLE
Adding deployment map source to SSM params used to identify out of date pipelines

### PR DESCRIPTION
**Pull request by StewartW**
*Issue #, if available:*

*Description of changes:*

Changed deployment params to include a separate prefix, allowing the "identify out
of date pipelines" function to be able to discriminate on deployment maps that
don't come from S3. (Isn't possible yet, but will make future expansion easier)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/pipeline_management/process_deployment_map.py [hunk 1]
```diff
@@ -74,6 +74,7 @@ def start_executions(sfn_client, deployment_map):
     )
     for pipeline in deployment_map.get("pipelines"):
         LOGGER.debug("Payload: %s", pipeline)
+        pipeline = {**pipeline, "deployment_map_source": "s3"}
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/lambda_codebase/pipeline_management/process_deployment_map.py:77]: `s3` is lower case here, while in the other hard coded paths they use `S3` instead.

**Comment by sbkok** [comment]
On the use of `--size-only`, this will break if you happen to replace part of your deployment map with the exact same number of characters.

For example, if you rename the pipeline from `vpc-secure` to `secure-vpc`, the size would be the same.
Same for changing the account id. I think this will happen more often than you expect with deployment maps.

**Comment by StewartW** [comment]
> On the use of `--size-only`, this will break if you happen to replace part of your deployment map with the exact same number of characters.
> 
> For example, if you rename the pipeline from `vpc-secure` to `secure-vpc`, the size would be the same. Same for changing the account id. I think this will happen more often than you expect with deployment maps.

Good point. I will remove size-only for now and look at an etag / versioned solution in the processing lambda function.

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 2]
```diff
@@ -769,8 +769,8 @@ Resources:
                 - pip install -r adf-build/requirements.txt -q -t ./adf-build
             build:
               commands:
-                - bash -c "[[ -e deployment_map.yml ]] && echo 'Copying deployment_map.yml' && aws s3 cp deployment_map.yml s3://$ADF_PIPELINES_BUCKET/deployment_map.yml || echo 'No deployment_map.yml, skipping copy'"
-                - bash -c "[[ -e deployment_maps ]] && echo 'Syncing deployment_maps folder' && aws s3 sync deployment_maps s3://$ADF_PIPELINES_BUCKET/deployment_maps || echo 'No deployment_maps folder, skipping sync'"
+                - bash -c "[[ -e deployment_map.yml ]] && echo 'Copying deployment_map.yml' && aws s3 cp deployment_map.yml s3://$ADF_PIPELINES_BUCKET/deployment_map.yml --size-only || echo 'No deployment_map.yml, skipping copy'"
```
[the remark this answers is not in the document]
Comment (StewartW) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml]: Doh

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 3]
```diff
@@ -770,7 +770,7 @@ Resources:
             build:
               commands:
                 - bash -c "[[ -e deployment_map.yml ]] && echo 'Copying deployment_map.yml' && aws s3 cp deployment_map.yml s3://$ADF_PIPELINES_BUCKET/deployment_map.yml || echo 'No deployment_map.yml, skipping copy'"
-                - bash -c "[[ -e deployment_maps ]] && echo 'Syncing deployment_maps folder' && aws s3 sync deployment_maps s3://$ADF_PIPELINES_BUCKET/deployment_maps || echo 'No deployment_maps folder, skipping sync'"
+                - bash -c "[[ -e deployment_maps ]] && echo 'Syncing deployment_maps folder' && aws s3 sync deployment_maps s3://$ADF_PIPELINES_BUCKET/deployment_maps --size-only --delete || echo 'No deployment_maps folder, skipping sync'"
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:773]: @StewartW Can you undo this line change?
We can rely on PR #530 to sync based on the hash instead if you agree with this approach.

The other changes still hold. Maybe we need to change the title of the PR though :smile:

**Review by sbkok — CHANGES_REQUESTED** [review]

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml (hunk 3 above)
Comment (StewartW) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:773]:
```suggestion
                - bash -c "[[ -e deployment_maps ]] && echo 'Syncing deployment_maps folder' && aws s3 sync deployment_maps s3://$ADF_PIPELINES_BUCKET/deployment_maps  || echo 'No deployment_maps folder, skipping sync'"
```

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml [hunk 4]
```diff
@@ -770,7 +770,7 @@ Resources:
             build:
               commands:
                 - bash -c "[[ -e deployment_map.yml ]] && echo 'Copying deployment_map.yml' && aws s3 cp deployment_map.yml s3://$ADF_PIPELINES_BUCKET/deployment_map.yml || echo 'No deployment_map.yml, skipping copy'"
-                - bash -c "[[ -e deployment_maps ]] && echo 'Syncing deployment_maps folder' && aws s3 sync deployment_maps s3://$ADF_PIPELINES_BUCKET/deployment_maps || echo 'No deployment_maps folder, skipping sync'"
+                - bash -c "[[ -e deployment_maps ]] && echo 'Syncing deployment_maps folder' && aws s3 sync deployment_maps s3://$ADF_PIPELINES_BUCKET/deployment_maps  || echo 'No deployment_maps folder, skipping sync'"
```
Comment (sbkok) [inline src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:773]: There is a single space that is causing this line to be marked as changed.
Could you remove that, such that the other PR will not create a conflict with this line change?

**Review by sbkok — CHANGES_REQUESTED** [review]

File: src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml (hunk 4 above)
Comment (StewartW) [reply src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml:773]:
```suggestion
                - bash -c "[[ -e deployment_maps ]] && echo 'Syncing deployment_maps folder' && aws s3 sync deployment_maps s3://$ADF_PIPELINES_BUCKET/deployment_maps || echo 'No deployment_maps folder, skipping sync'"
```

**Review by javydekoning — APPROVED** [review]

**Review by sbkok — APPROVED** [review]

**Pull request merged**